### PR TITLE
Configuration support

### LINKED ISSUE #3: Configuration
Nightwatch should be user-configurable.

As a start, let's make it able to load a bunch of config files from the directory that's passed as an argument. I choose YML as a configuration format, wire it up.

### PR DESCRIPTION
**Pull request by ForNeVeR**
Close #3.

TODO:

- [x] Add configuration file version

### CODE REVIEW

**Comment by ForNeVeR** [comment]
@gsomix wanna review?

File: Nightwatch/Configuration.fs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+open System.IO
+
+open YamlDotNet.Serialization
+open YamlDotNet.Serialization.NamingConventions
+
+open Nightwatch.FileSystem
+open Nightwatch.Resources
+
+[<CLIMutable>]
+type ResourceDescription =
```
Comment (ForNeVeR) [inline Nightwatch/Configuration.fs]: Make this `private`?

File: Nightwatch/Configuration.fs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+             let version = scalar.Value
+             box <| Version version
+        member __.WriteYaml(_, _, _) = failwithf "Not supported" }
+
+let private buildDeserializer() =
+    DeserializerBuilder()
+        .WithTypeConverter(versionConverter)
+        .Build()
+
+let private toResource = Result.map (fun { id = id; schedule = schedule; check = check } ->
```
Comment (gsomix) [inline Nightwatch/Configuration.fs]: I'd prefer 

```fsharp
Result.map <| fun res -> { id = res.id; runEvery = res.schedule; checkCommand = res.check }
```

File: Nightwatch/Program.fs [hunk 3]
```diff
@@ -3,17 +3,44 @@
...
+open Nightwatch.Configuration
+open Nightwatch.FileSystem
+open Nightwatch.Resources
+
 let private printVersion() =
     let version = Assembly.GetEntryAssembly().GetName().Version
     printfn "Nightwatch v. %A" version
+    printfn "Config file format v. %A" Configuration.configFormatVersion
+
+let private divideResults seq =
```
Comment (gsomix) [inline Nightwatch/Program.fs]: How about `groupResults`?

File: Nightwatch/Configuration.fs (hunk 1 above)
Comment (gsomix) [reply Nightwatch/Configuration.fs]: Yes, please.

File: Nightwatch/Program.fs (hunk 3 above)
Comment (ForNeVeR) [reply Nightwatch/Program.fs]: I think `split` is a better word.

File: Nightwatch/Configuration.fs (hunk 2 above)
Comment (ForNeVeR) [reply Nightwatch/Configuration.fs]: Yep, that'll be better. With one change: these days I rarely prefer to use `<|` operator, and that's not the case.

File: Nightwatch/Configuration.fs (hunk 1 above)
Comment (ForNeVeR) [reply Nightwatch/Configuration.fs]: Unfortunately, that's not possible: YamlDotNet wants a public type; otherwise it cannot find type's constructor.

**Comment by ForNeVeR** [comment]
Thanks for the review!

File: Nightwatch/Configuration.fs (hunk 1 above)
Comment (gsomix) [reply Nightwatch/Configuration.fs]: What about `[<RequireQualifiedAccess>]`?

Comment (ForNeVeR) [reply Nightwatch/Configuration.fs]: I don't think it works the way we need. Look: it should be applied to **container** (e.g. module or enum), and the container's contents won't be visible without qualification.

To achieve something with that I'd need to apply that attribute to the whole module, and I don't want this.

**Pull request merged**